core/dbt/parser/manifest.py

```diff
@@ -361,215 +559,27 @@ def load(self) -> Manifest:
-            self.manifest.rebuild_disabled_lookup()
-
-            # copy the selectors from the root_project to the manifest
-            self.manifest.selectors = self.root_project.manifest_selectors
-
-            # inject any available external nodes
-            self.manifest.build_parent_and_child_maps()
-            external_nodes_modified = self.inject_external_nodes()
-            if external_nodes_modified:
-                self.manifest.rebuild_ref_lookup()
-
-            # update the refs, sources, docs and metrics depends_on.nodes
-            # These check the created_at time on the nodes to
-            # determine whether they need processing.
-            start_process = time.perf_counter()
-            self.process_sources(self.root_project.project_name)
-            self.process_refs(self.root_project.project_name, self.root_project.dependencies)
-            self.process_docs(self.root_project)
-            self.process_metrics(self.root_project)
-            self.process_saved_queries(self.root_project)
-            self.check_valid_group_config()
-            self.check_valid_access_property()
-
-            semantic_manifest = SemanticManifest(self.manifest)
-            if not semantic_manifest.validate():
-                raise dbt.exceptions.ParsingError("Semantic Manifest validation failed.")
-
-            # update tracking data
-            self._perf_info.process_manifest_elapsed = time.perf_counter() - start_process
-            self._perf_info.static_analysis_parsed_path_count = (
-                self.manifest._parsing_info.static_analysis_parsed_path_count
-            )
-            self._perf_info.static_analysis_path_count = (
-                self.manifest._parsing_info.static_analysis_path_count
+            self.process_refs(
+                self.root_project.project_name,
+                self.root_project.dependencies,
             )
-
-        # Inject any available external nodes, reprocess refs if changes to the manifest were made.
-        external_nodes_modified = False
-        if skip_parsing:
-            # If we didn't skip parsing, this will have already run because it must run
-            # before process_refs. If we did skip parsing, then it's possible that only
-            # external nodes have changed and we need to run this to capture that.
-            self.manifest.build_parent_and_child_maps()
-            external_nodes_modified = self.inject_external_nodes()
-            if external_nodes_modified:
-                self.manifest.rebuild_ref_lookup()
-                self.process_refs(
-                    self.root_project.project_name,
-                    self.root_project.dependencies,
-                )
-                # parent and child maps will be rebuilt by write_manifest
-
-        if not skip_parsing or external_nodes_modified:
+            # parent and child maps will be rebuilt by write_manifest
+        if not self.skip_parsing or external_nodes_modified:
```

Comment:
I don't think this writing out manifest should belong to this function given the function name.
If we move it out to the load layer then I think we don't have to add the global attribute `skip_parsing`?